### PR TITLE
Add crd for bigquerydataset

### PR DESCRIPTION
**Pull request by xiaoweim**
Following the path to add a new config connector resource BigQueryDatasetAccess https://docs.google.com/document/d/1Az2yOr9dGHrj-IGvYEhGxjpajGmcd65mQ4OjsAQSXyE/edit?pli=1&resourcekey=0-Mf91G-_QDqq6XEVzf_CnQQ&tab=t.0#heading=h.69lvxccu0hq2

This is the first step to add CRD/API for BigQueryDataset since BigQueryDatasetAccess only represents a single entry in the BigQueryDataset spec.access field.

### CODE REVIEW

File: dev/tasks/generate-crds [hunk 1]
```diff
@@ -55,7 +55,9 @@ go run ./scripts/crd-tools reflow-descriptions --dir apis/config/crd/
 
 # excluded_resources are resources that are under development for a direct conversion
 # we don't modify the CRD just yet for those but will in the future
-excluded_resources=("computeforwardingrule" "bigquerydatatransferconfig")
```
Comment (yuwenma) [inline dev/tasks/generate-crds:58]: We do not want to use this flag if the resource is already Beta.  https://docs.google.com/document/d/1Az2yOr9dGHrj-IGvYEhGxjpajGmcd65mQ4OjsAQSXyE/edit?resourcekey=0-Mf91G-_QDqq6XEVzf_CnQQ&tab=t.0#heading=h.8mqswg27ngyc

Comment (xiaoweim) [reply dev/tasks/generate-crds]: Found the instruction to resolve this!

File: dev/tools/proto-to-mapper/Makefile [hunk 2]
```diff
@@ -14,6 +14,7 @@ generate-pb: install-protoc-linux
 		./third_party/googleapis/google/api/*.proto \
 		./third_party/googleapis/google/api/*/*/*.proto \
 		./third_party/googleapis/google/bigtable/*/*/*.proto \
+		./third_party/googleapis/google/cloud/bigquery/v2/*.proto \
```
Comment (yuwenma) [inline dev/tools/proto-to-mapper/Makefile:17]: nit: Suggest match all versions "./third_party/googleapis/google/cloud/bigquery/*/*.proto"

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ metadata:
   creationTimestamp: null
   labels:
     cnrm.cloud.google.com/managed-by-kcc: "true"
-    cnrm.cloud.google.com/stability-level: stable
+    cnrm.cloud.google.com/stability-level: alpha
```
Comment (yuwenma) [inline config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:9]: This diff means you need to add the stability-level tag in the API. Here's an example https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/8fe8c6dd64cc1034dfb2ab00105bf258dc885406/apis/kms/v1beta1/kmskeyring.go#L73

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml [hunk 4]
```diff
@@ -16,10 +16,8 @@ spec:
     categories:
     - gcp
     kind: BigQueryDataset
+    listKind: BigQueryDatasetList
     plural: bigquerydatasets
-    shortNames:
-    - gcpbigquerydataset
-    - gcpbigquerydatasets
```
Comment (yuwenma) [inline config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:22]: You need to update the go tag in apis to include these shortnames, example https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/8fe8c6dd64cc1034dfb2ab00105bf258dc885406/apis/kms/v1beta1/kmskeyring.go#L71

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml [hunk 5]
```diff
@@ -92,46 +99,55 @@ spec:
...
-                      description: An email address of a Google Group to grant access
-                        to.
+                      description: '[Pick one] An email address of a Google Group
+                        to grant access to. Maps to IAM policy member "group:GROUP".'
                       type: string
                     iamMember:
-                      description: |-
-                        Some other type of member that appears in the IAM Policy but isn't a user,
-                        group, domain, or special group. For example: 'allUsers'.
+                      description: '[Pick one] Some other type of member that appears
```
Comment (yuwenma) [inline config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:112]: Do you happen to know where this `[Pick one]` comes from? Very interesting tag.

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml [hunk 6]
```diff
@@ -226,86 +242,71 @@ spec:
...
               maxTimeTravelHours:
-                description: Defines the time travel window in hours. The value can
-                  be from 48 to 168 hours (2 to 7 days).
+                description: Optional. Defines the time travel window in hours. The
+                  value can be from 48 to 168 hours (2 to 7 days). The default value
+                  is 168 hours if this is not set.
                 type: string
               projectRef:
-                description: The project that this resource belongs to.
+                description: The project that this resource belongs to. optional.
```
Comment (yuwenma) [inline config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:309]: This seems to be wrong. The projectRef is required.

File: apis/bigquery/v1beta1/bigquerydataset_types.go [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+	//  locations.
+	Location *string `json:"location,omitempty"`
+
+	// Optional. Defines the time travel window in hours. The value can be from 48
+	//  to 168 hours (2 to 7 days). The default value is 168 hours if this is not
+	//  set.
+	MaxTimeTravelHours *string `json:"maxTimeTravelHours,omitempty"`
+
+	//	The project that this resource belongs to.
+	//  optional.
```
Comment (yuwenma) [inline apis/bigquery/v1beta1/bigquerydataset_types.go:106]: nit: remove "optional"

**Review by yuwenma — COMMENTED** [review]
/approve

Some nits, overall looks great!

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml (hunk 6 above)
Comment (xiaoweim) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:309]: From our documentation, the projectRef under Spec seems to be optional https://cloud.google.com/config-connector/docs/reference/resource-docs/bigquery/bigquerydataset#spec

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml (hunk 5 above)
Comment (xiaoweim) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:112]: This is from this proto https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/v2/dataset.proto#L172

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml (hunk 6 above)
Comment (yuwenma) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:312]: Do you mind verifying that what happens if the `spec.projectRef` is not provided against the real GCP?

Comment (xiaoweim) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_bigquerydatasets.bigquery.cnrm.cloud.google.com.yaml:312]: We currently have two testcases that created bigquerydataset without a projectRef, I have run it against real GCP and working.
Also verified with a real CC cluster, created a bigquerydataset without projectRef, with only name and namespace through `kubectl apply`, and it's status is upToDate Ready.
Bigquerydataset yaml file.
```
apiVersion: bigquery.cnrm.cloud.google.com/v1beta1
kind: BigQueryDataset
metadata:
  name: bigquerydatasetsampletest
  namespace: config-control
spec:
  friendlyName: bigquerydataset-sample
```
Resource status:
```
$ kubectl get bigquerydataset -n config-control
NAME                        AGE   READY   STATUS     STATUS AGE
bigquerydatasetsampletest   97s   True    UpToDate   96s
```

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

**Comment by yuwenma** [comment]
Looks great! Thank you

**Pull request merged**